### PR TITLE
fdhj2b beadandó

### PR DESCRIPTION
**Pull request by mviktor02**

### CODE REVIEW

File: FDHJ2B/beadando/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,14 @@
+data "google_project" "project" {
```
Comment (s1mark) [inline FDHJ2B/beadando/outputs.tf:1]: please put the datasource type for example in the main.tf 
the outputs.tf should only contain output types

**Comment by s1mark** [comment]
Nice and complex solution
Have you tried running terraform init and terraform plan?

Could you share the plan out?

**Comment by mviktor02** [comment]
terraform plan output:
```
data.google_project.project: Reading...
module.vm.module.sa.module.service_accounts.data.template_file.keys["fdhj2b"]: Reading...
module.vm.module.sa.module.service_accounts.data.template_file.keys["fdhj2b"]: Read complete after 0s [id=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
data.google_project.project: Read complete after 1s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # module.vm.google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-cloud/debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "fdhj2b"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # module.vm.google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "fdhj2b"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + attached_disk {
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)
        }

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # module.vm.module.api.module.google_project_service.google_project_service.project_services["compute.googleapis.com"] will be created
  + resource "google_project_service" "project_services" {
      + disable_dependent_services = true
      + disable_on_destroy         = true
      + id                         = (known after apply)
      + project                    = "ekke-cloud"
      + service                    = "compute.googleapis.com"
    }

  # module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"] will be created
  + resource "google_project_service" "project_services" {
      + disable_dependent_services = true
      + disable_on_destroy         = true
      + id                         = (known after apply)
      + project                    = "ekke-cloud"
      + service                    = "iam.googleapis.com"
    }

  # module.vm.module.sa.module.service_accounts.google_project_iam_member.project-roles["fdhj2b-ekke-cloud=>roles/monitoring.admin"] will be created
  + resource "google_project_iam_member" "project-roles" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "ekke-cloud"
      + role    = "roles/monitoring.admin"
    }

  # module.vm.module.sa.module.service_accounts.google_service_account.service_accounts["fdhj2b"] will be created
  + resource "google_service_account" "service_accounts" {
      + account_id   = "fdhj2b"
      + disabled     = false
      + display_name = "Terraform-managed service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = "ekke-cloud"
      + unique_id    = (known after apply)
    }

Plan: 6 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "ekke-cloud"
  + project_number = "92227481840"
  + sa_email       = (known after apply)
```

**Review by s1mark — CHANGES_REQUESTED** [review]
Good job so far. Only the file creation is left I think

**Comment by FlashHUN** [comment]
> Good job so far. Only the file creation is left I think

https://github.com/s1mark/cloud/blob/a6ad57c58a50169dd7639e2951c8d5e9eb9f50b3/FDHJ2B/beadando/vm/main.tf#L21

**Review by s1mark — APPROVED** [review]

**Comment by s1mark** [comment]
The last thing would be to apply it and if it all went through a do a destroy.
Good job

**Comment by mviktor02** [comment]
`terraform apply` output:
```
data.google_project.project: Reading...
module.vm.module.api.module.google_project_service.google_project_service.project_services["compute.googleapis.com"]: Refreshing state... [id=ekke-cloud/compute.googleapis.com]
module.vm.module.sa.module.service_accounts.data.template_file.keys["fdhj2b"]: Reading...
module.vm.module.sa.module.service_accounts.data.template_file.keys["fdhj2b"]: Read complete after 0s [id=e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855]
data.google_project.project: Read complete after 2s [id=projects/ekke-cloud]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # module.vm.google_compute_disk.default will be created
  + resource "google_compute_disk" "default" {
      + creation_timestamp        = (known after apply)
      + id                        = (known after apply)
      + image                     = "debian-cloud/debian-11-bullseye-v20220719"
      + label_fingerprint         = (known after apply)
      + last_attach_timestamp     = (known after apply)
      + last_detach_timestamp     = (known after apply)
      + name                      = "fdhj2b-attached-disk"
      + physical_block_size_bytes = 4096
      + project                   = (known after apply)
      + provisioned_iops          = (known after apply)
      + self_link                 = (known after apply)
      + size                      = 10
      + source_disk_id            = (known after apply)
      + source_image_id           = (known after apply)
      + source_snapshot_id        = (known after apply)
      + type                      = "pd-standard"
      + users                     = (known after apply)
      + zone                      = "europe-central2-a"
    }

  # module.vm.google_compute_instance.default will be created
  + resource "google_compute_instance" "default" {
      + can_ip_forward          = false
      + cpu_platform            = (known after apply)
      + current_status          = (known after apply)
      + deletion_protection     = false
      + guest_accelerator       = (known after apply)
      + id                      = (known after apply)
      + instance_id             = (known after apply)
      + label_fingerprint       = (known after apply)
      + machine_type            = "f1-micro"
      + metadata_fingerprint    = (known after apply)
      + metadata_startup_script = "echo done > ~/task"
      + min_cpu_platform        = (known after apply)
      + name                    = "fdhj2b"
      + project                 = (known after apply)
      + self_link               = (known after apply)
      + tags_fingerprint        = (known after apply)
      + zone                    = "europe-central2-a"

      + attached_disk {
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)
        }

      + boot_disk {
          + auto_delete                = true
          + device_name                = (known after apply)
          + disk_encryption_key_sha256 = (known after apply)
          + kms_key_self_link          = (known after apply)
          + mode                       = "READ_WRITE"
          + source                     = (known after apply)

          + initialize_params {
              + image  = "debian-cloud/debian-11"
              + labels = (known after apply)
              + size   = (known after apply)
              + type   = (known after apply)
            }
        }

      + confidential_instance_config {
          + enable_confidential_compute = (known after apply)
        }

      + network_interface {
          + ipv6_access_type   = (known after apply)
          + name               = (known after apply)
          + network            = "default"
          + network_ip         = (known after apply)
          + stack_type         = (known after apply)
          + subnetwork         = (known after apply)
          + subnetwork_project = (known after apply)
        }

      + reservation_affinity {
          + type = (known after apply)

          + specific_reservation {
              + key    = (known after apply)
              + values = (known after apply)
            }
        }

      + scheduling {
          + automatic_restart           = (known after apply)
          + instance_termination_action = (known after apply)
          + min_node_cpus               = (known after apply)
          + on_host_maintenance         = (known after apply)
          + preemptible                 = (known after apply)
          + provisioning_model          = (known after apply)

          + node_affinities {
              + key      = (known after apply)
              + operator = (known after apply)
              + values   = (known after apply)
            }
        }

      + service_account {
          + email  = (known after apply)
          + scopes = [
              + "https://www.googleapis.com/auth/monitoring",
            ]
        }
    }

  # module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"] will be created
  + resource "google_project_service" "project_services" {
      + disable_dependent_services = true
      + disable_on_destroy         = true
      + id                         = (known after apply)
      + project                    = "ekke-cloud"
      + service                    = "iam.googleapis.com"
    }

  # module.vm.module.sa.module.service_accounts.google_project_iam_member.project-roles["fdhj2b-ekke-cloud=>roles/monitoring.admin"] will be created      
  + resource "google_project_iam_member" "project-roles" {
      + etag    = (known after apply)
      + id      = (known after apply)
      + member  = (known after apply)
      + project = "ekke-cloud"
      + role    = "roles/monitoring.admin"
    }

  # module.vm.module.sa.module.service_accounts.google_service_account.service_accounts["fdhj2b"] will be created
  + resource "google_service_account" "service_accounts" {
      + account_id   = "fdhj2b"
      + disabled     = false
      + display_name = "Terraform-managed service account"
      + email        = (known after apply)
      + id           = (known after apply)
      + member       = (known after apply)
      + name         = (known after apply)
      + project      = "ekke-cloud"
      + unique_id    = (known after apply)
    }

Plan: 5 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + project_id     = "ekke-cloud"
  + project_number = "92227481840"
  + sa_email       = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"]: Creating...
module.vm.module.sa.module.service_accounts.google_service_account.service_accounts["fdhj2b"]: Creating...
module.vm.google_compute_disk.default: Creating...
module.vm.module.sa.module.service_accounts.google_service_account.service_accounts["fdhj2b"]: Creation complete after 2s [id=projects/ekke-cloud/serviceAccounts/fdhj2b@ekke-cloud.iam.gserviceaccount.com]
module.vm.module.sa.module.service_accounts.google_project_iam_member.project-roles["fdhj2b-ekke-cloud=>roles/monitoring.admin"]: Creating...
module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"]: Still creating... [10s elapsed]
module.vm.google_compute_disk.default: Still creating... [10s elapsed]
module.vm.google_compute_disk.default: Creation complete after 13s [id=projects/ekke-cloud/zones/europe-central2-a/disks/fdhj2b-attached-disk]
module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"]: Still creating... [20s elapsed]
module.vm.module.api.module.google_project_service.google_project_service.project_services["iam.googleapis.com"]: Creation complete after 24s [id=ekke-cloud/iam.googleapis.com]
╷
│ Error: Request `Create IAM Members roles/monitoring.admin serviceAccount:fdhj2b@ekke-cloud.iam.gserviceaccount.com for project "ekke-cloud"` returned error: Error 
applying IAM policy for project "ekke-cloud": Error setting IAM policy for project "ekke-cloud": googleapi: Error 403: Policy update access denied., forbidden       
│
│   with module.vm.module.sa.module.service_accounts.google_project_iam_member.project-roles["fdhj2b-ekke-cloud=>roles/monitoring.admin"],
│   on .terraform\modules\vm.sa.service_accounts\main.tf line 46, in resource "google_project_iam_member" "project-roles":
│   46: resource "google_project_iam_member" "project-roles" {
│
```

I'm guessing I don't have permission to update the IAM policy at the moment. Everything else looks fine, also ran `terraform destroy` right after apply.

**Pull request merged**